File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> newHash = new HashMap<>();
        int n = nums.length;
        int[] output = new int[2];

        for(int i=0; i<n; i++){
            int comp = target - nums[i];

            if(newHash.containsKey(comp)){
                output[0] = newHash.get(comp);
                output[1] = i;
                return output;
            }

            newHash.put(nums[i], i);
        }

        return new int[] {};
    }
}

File: 1-two-sum/two-sum.js
function twoSum(nums, target){

    let obj = {}
    let ans = []

    for(let i=0; i<nums.length; i++){
        let rem = target - nums[i];

        if(rem in obj){
            ans[0] = i;
            ans[1] = obj[rem]
            break
        }

        obj[nums[i]] = i;
    }

    return ans;
}